x/stake/keeper/validator.go

```diff
@@ -99,42 +99,28 @@ func (k Keeper) validatorByPowerIndexExists(ctx sdk.Context, power []byte) bool
 	return store.Get(power) != nil
 }
 
-// Get the set of all validators with no limits, used during genesis dump
-func (k Keeper) GetAllValidators(ctx sdk.Context) (validators []types.Validator) {
-	store := ctx.KVStore(k.storeKey)
-	iterator := sdk.KVStorePrefixIterator(store, ValidatorsKey)
-
-	i := 0
-	for ; ; i++ {
-		if !iterator.Valid() {
-			break
-		}
-		addr := iterator.Key()[1:]
-		validator := types.MustUnmarshalValidator(k.cdc, addr, iterator.Value())
-		validators = append(validators, validator)
-		iterator.Next()
+// Get the set of all validators. If maxRetrieve is supplied, the respective amount will be returned.
+func (k Keeper) GetValidators(ctx sdk.Context, maxRetrieve ...int16) (validators []types.Validator) {
+	retrieve := len(maxRetrieve) > 0
+	if retrieve {
```

Comment:
we shouldn't allow for retrieve all to exist - meaning that this retrieve term should be taken out - see my comments on the old PR